src/user/user.spec.js

```diff
@@ -80,6 +80,102 @@ describe("user", () => {
     });
   });
 
+  describe(".set() :", () => {
+    withData(
+      {
+        "sets an email": { email: "wayne_modified@vates.fr" },
+        "sets a password": { password: "newPassword" },
+        "sets a permission": { permission: "user" },
+        "sets a preference": {
+          preferences: {
+            filters: {
+              VM: {
+                test: "name_label: test",
+              },
+            },
+          },
+        },
+      },
+      async data => {
+        data.id = await xo.createUser(simpleUser);
+        expect(await xo.call("user.set", data)).toBe(true);
+        expect(omit(await xo.getUser(data.id), "id")).toMatchSnapshot();
+
+        await testConnection({
+          credentials: {
+            email: data.email === undefined ? simpleUser.email : data.email,
+            password:
+              data.password === undefined ? simpleUser.password : data.password,
+          },
+        });
+      }
+    );
+
+    withData(
+      {
+        "fails trying to set an email with a non admin user connection": {
+          email: "wayne_modified@vates.fr",
+        },
+        "fails trying to set a password with a non admin user connection": {
+          password: "newPassword",
+        },
+        "fails trying to set a permission with a non admin user connection": {
+          permission: "user",
+        },
+      },
+      async data => {
+        data.id = await xo.createUser({
+          email: "wayne8@vates.fr",
+          password: "batman8",
+        });
+        await xo.createUser(simpleUser);
+
+        await testWithOtherConnection(simpleUser, xo =>
+          expect(xo.call("user.set", data)).rejects.toMatchSnapshot()
+        );
+      }
+    );
+
+    it("fails trying to set user permission with its own connection", async () => {
+      const userId = await xo.createUser(simpleUser);
+      await testWithOtherConnection(simpleUser, xo =>
+        expect(
+          xo.call("user.set", { id: userId, permission: "user" })
+        ).rejects.toMatchSnapshot()
+      );
+    });
+
+    it("fails trying to change its own permission as an admin", async () => {
+      await expect(
+        xo.call("user.set", { id: xo._user.id, permission: "user" })
```

Comment:
It can cause troubles if this call not reject and the `tester` has an only admin account.